checkov/gitlab_ci/runner.py

```diff
@@ -0,0 +1,83 @@
+from checkov.common.images.image_referencer import ImageReferencer
+from checkov.common.output.report import CheckType
+from checkov.gitlab_ci.checks.registry import registry
+from checkov.yaml_doc.runner import Runner as YamlRunner
+
+class Runner(YamlRunner, ImageReferencer):
+    check_type = CheckType.GITLAB_CI
+
+    def __init__(self):
+        super().__init__()
+
+    def require_external_checks(self):
+        return False
+
+    def import_registry(self):
+        return registry
+
+    def _parse_file(self, f):
+        if self.is_workflow_file(f):
+            return super()._parse_file(f)
+
+    def is_workflow_file(self, file_path):
+        """
+        :return: True if the file mentioned is in the gitlab workflow name .gitlab-ci.yml. Otherwise: False
+        """
+        return file_path.endswith((".gitlab-ci.yml",".gitlab-ci.yaml"))
+
+    def get_images(self, file_path):
+        """
+        Get container images mentioned in a file
+        :param file_path: File to be inspected
+        GitLab a workflow file can have a job and services run within a container.
+
+        in the following sample file we can see a node:14.16 image:
+
+        default:
+            image:
+                name: ruby:2.6
+                entrypoint: ["/bin/bash"]
+
+            image: nginx:1.18
+
+            services:
+                - name: privateregistry/stuff/my-postgres:11.7
+                  alias: db-postgres
+                - name: redis:latest  
+                - nginx:1.17
+        Source: https://docs.gitlab.com/ee/ci/docker/using_docker_images.html
+
+        :return: List of container image short ids mentioned in the file.
+        Example return value for a file with node:14.16 image: ['sha256:6a353e22ce']
+        """
+
+        images = set()
+        imagesKeys = ["image","services"]
+        workflow, workflow_line_numbers = self._parse_file(file_path)
+
+        for job_name, job_object in workflow.items():
+            if isinstance(job_object, dict):
+                for subjob in job_object.items():
```

Comment:
```suggestion
                for key, subjob in job_object.items():
```
then the rest should be easier, because you created `subjob` as a tuple, then you don't need to do `subjob[1]`

Reply:
Smarter.  Thanks.